code/modules/reagents/newchem/newchem_pyro.dm

```diff
@@ -0,0 +1,174 @@
+	if(!istype(M, /mob/living))
+		return
+	if(method == TOUCH)
+		var/turf/simulated/T = get_turf(M)
+		goonchem_vortex(T, 1, 5, 3)
+
+
+/datum/chemical_reaction/sorium/on_reaction(var/datum/reagents/holder, var/created_volume)
+	var/turf/simulated/T = get_turf(holder.my_atom)
+	goonchem_vortex(T, 1, 5, 6)
+
+/datum/reagent/liquid_dark_matter
+	name = "Liquid Dark Matter"
+	id = "liquid_dark_matter"
+	description = "Sucks everything into the detonation point."
+	reagent_state = LIQUID
+	color = "#60A584"  //rgb: 96, 165, 132
+
+/datum/chemical_reaction/liquid_dark_matter
+	name = "Liquid Dark Matter"
+	id = "liquid_dark_matter"
+	required_reagents = list("stable_plasma" = 1, "radium" = 1, "carbon" = 1)
+	min_temperature = 474
+	results = list("liquid_dark_matter" = 3)
+
+/datum/reagent/liquid_dark_matter/reaction_turf(var/turf/simulated/T, var/volume)
+	if(istype(T, /turf/simulated/floor/))
+		goonchem_vortex(T, 0, 5, 3)
+		return
+/datum/reagent/liquid_dark_matter/reaction_mob(var/mob/living/M, var/method=TOUCH, var/volume)
+	if(!istype(M, /mob/living))
+		return
+	if(method == TOUCH)
+		var/turf/simulated/T = get_turf(M)
+		goonchem_vortex(T, 0, 5, 3)
+		return
+/datum/chemical_reaction/liquid_dark_matter/on_reaction(var/datum/reagents/holder, var/created_volume)
+	var/turf/simulated/T = get_turf(holder.my_atom)
+	goonchem_vortex(T, 0, 5, 6)
+	return
+
+
+/proc/goonchem_vortex(var/turf/simulated/T, var/setting_type, var/range, var/pull_times)
+	for(var/atom/movable/X in orange(range, T))
+		if(istype(X, /obj/effect))
+			continue  //stop pulling smoke and hotspots please
+		if(istype(X, /atom/movable))
+			if((X) && !X.anchored)
+				if(setting_type)
+					for(var/i = 0, i < pull_times, i++)
+						step_away(X,T)
+				else
+					for(var/i = 0, i < pull_times, i++)
+						step_towards(X,T)
+
+
+/datum/reagent/blackpowder
+	name = "Black Powder"
+	id = "blackpowder"
+	description = "Explodes. Violently."
+	reagent_state = LIQUID
+	color = "#000000"  //rgb: 96, 165, 132
+	custom_metabolism = 0.05
+
+/datum/chemical_reaction/blackpowder
+	name = "Black Powder"
+	id = "blackpowder"
+	required_reagents = list("saltpetre" = 1, "charcoal" = 1, "sulfur" = 1)
+	results = list("blackpowder" = 3)
+
+/datum/chemical_reaction/blackpowder_explosion
+	name = "Black Powder Kaboom"
+	id = "blackpowder_explosion"
+	required_reagents = list("blackpowder" = 1)
+	min_temperature = 474
+	results = list(null = 1)
+
+/datum/chemical_reaction/blackpowder_explosion/on_reaction(var/datum/reagents/holder, var/created_volume)
+	holder.my_atom.visible_message("<span class = 'userdanger'>Sparks come out of [holder.my_atom]!</span>")
+	sleep(rand(10,30))
+	var/turf/simulated/T = get_turf(holder.my_atom)
+	var/ex_severe = round(created_volume / 10)
+	var/ex_heavy = round(created_volume / 8)
+	var/ex_light = round(created_volume / 6)
+	var/ex_flash = round(created_volume / 4)
+	if(ex_severe > 3)
+		ex_severe = 3
+	if(ex_heavy > 7)
+		ex_severe = 7
```

Comment:
Hell, get rid of the ifs entirely.
They should just be ex_whatever = Clamp(round(created_volume / #), 0, ex_max)